// @vonage/server-client/dist/lib/enums/AuthenticationType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthenticationType = void 0;
/**
 * Enum representing the different types of authentication methods
 * supported by the Vonage APIs.
 */
var AuthenticationType;
(function (AuthenticationType) {
    /**
     * Basic authentication using a base64-encoded string.
     */
    AuthenticationType["BASIC"] = "basic";
    /**
     * JSON Web Token (JWT) authentication.
     */
    AuthenticationType["JWT"] = "jwt";
    /**
     * Authentication using both API key and secret in the request body.
     * @deprecated This method is deprecated.
     */
    AuthenticationType["KEY_SECRET"] = "key_secret";
    /**
     * Authentication using API key and secret in the query parameters.
     */
    AuthenticationType["QUERY_KEY_SECRET"] = "query_key_secret";
    /**
     * HMAC signature-based authentication.
     */
    AuthenticationType["SIGNATURE"] = "signature";
    /**
     * CIBA authentication
     */
    AuthenticationType["CIBA"] = "ciba";
    /**
     * OAuth2 authentication.
     */
    AuthenticationType["OAUTH2"] = "oauth2";
})(AuthenticationType || (exports.AuthenticationType = AuthenticationType = {}));
